src/lib/consenthandler/drivers/tcf.js

```diff
@@ -10,19 +10,29 @@ export const TCF_API_VERSION = 2;
 export class Tcf extends BaseCmp {
     constructor() {
         super(TCF_API, TCF_RETURN_MSG, TCF_GET_MSG, TCF_FRAME);
-        this.tcData = {};
+        this.cmpData = {};
         this.consentCallbackList = [];
     }
 
     /**
-     * maps the fields from the api calls to a common result set
-     * @param result
-     * @returns {{hasSiteConsent: *, gdpr_consent: *, gdpr: *}}
+     * Normalize CMP v2 tcData into an object.
+     * @param tcData A tcData from v2 requests.
+     * @return {Object} Result with normalized fields.
      */
-    getConsentData(result) {
-        if(!result.publisher){ result.publisher = {}; }
-        if(!result.publisher.consents){ result.publisher.consents = {}; }
-        return this.formatConsentData(result.gdprApplies, result.tcString, result.publisher.consents['1']);
+
+    formatData(tcData) {
+        const ret = {version: 2};
+        if (tcData) {
+            ret.gdprApplies = tcData.gdprApplies;
+            ret.consentString = tcData.tcString;
+            ret.tcData = tcData;
```

Comment:
I thought the goal was to abstract the different drivers away, if you're returning tcData that means the caller has to know what driver is being used and handle the result differently

Reply:
Yes, we abstract what we can, but also provide the raw data so caller can make their own evaluation if needed.

Reply:
then do you want to return results for cmp instead of just the vendor data

Reply:
Isn't tcData the result for TCFv2?

Reply:
yes, I was thinking of results for cmp